shacl12-node-expr/index.html

```diff
@@ -557,17 +558,177 @@ <h3>Document Conventions</h3>
+	sh:path ex:employee ;
+	sh:class ex:Person .
+
+ex:Company-employeeCount
+	a sh:PropertyShape ;
+	sh:name "employee count" ;
+	sh:description "The number of employees, automatically computed." ;
+	sh:path ex:employeeCount ;
+	sh:datatype xsd:integer ;
+	sh:values <b>[
+		sh:count [
+			sh:path ex:employee ;
+		]
+	]</b> .
+						</div>
+						<div class="jsonld">
+							TODO
+						</div>
+					</div>
+				</aside>
+			</p>
+			<div>
+				<img 
+				alt="Illustration of the data flow between node expressions computing the employee count"
+				src="images/CountExampleDiagram.png"
+				width="800px"
+				>
+			</div>
+			<p>
+				A difference between this example and the previous examples about <code>sh:targetNode</code>
+				is that these node expressions are evaluated against a given <a>focus node</a>.
+				So when a data visualization needs to render an instance of <code>ex:Company</code>,
+				the currently displayed company is the <a>focus node</a>, from which the number of employees
+				will be fetched.
+			</p>
+			<p class="todo">Clarify when these derived properties can be used (e.g. in sh:path expressions but not as triples elsewhere)</p>
+		</section>
+
 		<section id="syntax">
 			<h2>Node Expression Syntax</h2>
 			<p>
 				This section introduces the general syntax of SHACL <a>node expressions</a>.
 			</p>
+			<p>
+				The term <a>node expression function</a> refers to the <em>kind</em> or <em>type</em>
+				of a <a>node expression</a>.
+				For example, <code>sh:FilterShapeExpression</code> is a <a>node expression function</a>
+				while a specific instance of this function in the <a>graph</a> is the <a>node expression</a> itself.
+			</p>
+			<p>
+				The most basic node expression functions are <a>constant node expressions</a>, which are either
+				<a>literals</a> or <a>IRIs</a> and simply evaluate to these constants.
+				All other node expressions are represented by <a>blank nodes</a> and come in the following two variations.
+			</p>
+			<ul>
+				<li>
+					<a>Named parameter functions</a> are represented by a <a>blank node</a> that is the subject of one or more
+					triples, including the <a>key parameter</a> of the node expression function.
+				</li>
+				<li>
+					<a>List parameter functions</a> are comparable to traditional functions in, for example, SPARQL
+					and are represented by a <a>blank node</a> that is the subject of a single triple with a SHACL list as its object.
+				</li>
+			</ul>
 
 			<section id="ConstantNodeExpression">
 				<h3>Constant Node Expressions</h3>
 				<p>
-					The <a>node expression functions</a> in this section were introduced in the SHACL Core
-					specification but are repeated here to keep this document self-contained.
+					The <a>node expression functions</a> in this section are called <dfn>constant node expressions</dfn>.
+					They were introduced in the SHACL Core specification but are repeated here to keep this document self-contained.
```

Comment:
I wonder whether this would be better set off as a NOTE to the reader?
```suggestion
					They were introduced in the SHACL Core specification and are repeated here to keep this document self-contained.
```